### PR TITLE
Add new e2e testcases for security policy

### PR DESCRIPTION
**Pull request by songrx1997**
/assign @swetharepakula

### CODE REVIEW

**Comment by songrx1997** [comment]
cc @gauravkghildiyal

File: cmd/e2e-test/security_policy_test.go [hunk 1]
```diff
@@ -262,6 +263,210 @@ func TestSecurityPolicyTransition(t *testing.T) {
...
+			t.Fatalf("Error getting test backend config: %v", err)
+		}
+		testBackendConfig.Spec.SecurityPolicy = nil
+		testBackendConfig, err = bcCRUD.Update(testBackendConfig)
+		if err != nil {
+			t.Fatalf("Error updating test backend config: %v", err)
+		}
+		t.Logf("Backend config %s/%s updated", s.Namespace, testBackendConfig.Name)
+
+		// Wait for backend config Reconciliation.
```
Comment (swetharepakula) [inline cmd/e2e-test/security_policy_test.go:347]: nit: expand the comment to state we are expected no change, so waiting a certain period.

File: cmd/e2e-test/security_policy_test.go [hunk 2]
```diff
@@ -262,6 +263,210 @@ func TestSecurityPolicyTransition(t *testing.T) {
...
+		time.Sleep(transitionPollTimeout)
+
+		// Verify current backend status.
```
Comment (swetharepakula) [inline cmd/e2e-test/security_policy_test.go:350]: nit: add comment that we expect that the security policy was not removed.

File: cmd/e2e-test/security_policy_test.go [hunk 3]
```diff
@@ -262,6 +263,210 @@ func TestSecurityPolicyTransition(t *testing.T) {
...
+				t.Fatalf("fuzz.GCLBForVIP(..., %q, %q) = _, %v; want _, nil", vip, features.SecurityPolicy, err)
+			}
+
+			if err := verifySecurityPolicy(t, gclb, s.Namespace, testSvc.Name, ""); err != nil {
+				t.Logf("verifySecurityPolicy(..., %q, %q, %q) = %v, want nil", s.Namespace, testSvc.Name, "", err)
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			t.Errorf("Failed to wait for security policy updated: %v", err)
```
Comment (swetharepakula) [inline cmd/e2e-test/security_policy_test.go:459]: nit: "Failed waiting for security policy to be updated on backend service"

**Review by swetharepakula — COMMENTED** [review]
a couple of nits on the comments and error messages, otherwise LGTM

**Pull request closed without merging**